cli/flags/global.go

```diff
@@ -0,0 +1,275 @@
+		}),
+		GenericWithDeprecatedFlag(opts, &cli.GenericFlag[string]{
+			Name:    LogCustomFormatFlagName,
+			EnvVars: EnvVars(LogCustomFormatFlagName),
+			Usage:   "Set the custom log formatting.",
+			Setter:  opts.LogFormatter.SetCustomFormat,
+		}),
+		BoolWithDeprecatedFlag(opts, &cli.BoolFlag{
+			Name:        NonInteractiveFlagName,
+			EnvVars:     EnvVars(NonInteractiveFlagName),
+			Destination: &opts.NonInteractive,
+			Usage:       `Assume "yes" for all prompts.`,
+		}),
+
+		// Experiment Mode flags
+
+		BoolWithDeprecatedFlag(opts, &cli.BoolFlag{
+			Name:    ExperimentModeFlagName,
+			EnvVars: EnvVars(ExperimentModeFlagName),
+			Usage:   "Enables experiment mode for Terragrunt. For more information, see https://terragrunt.gruntwork.io/docs/reference/experiment-mode .",
+			Setter: func(_ bool) error {
+				opts.Experiments.ExperimentMode()
+
+				return nil
+			},
+		}),
+		SliceWithDeprecatedFlag(opts, &cli.SliceFlag[string]{
+			Name:    ExperimentFlagName,
+			EnvVars: EnvVars(ExperimentFlagName),
+			Usage:   "Enables specific experiments. For a list of available experiments, see https://terragrunt.gruntwork.io/docs/reference/experiment-mode .",
+			Setter:  opts.Experiments.EnableExperiment,
+			Action: func(_ *cli.Context, val []string) error {
+				opts.Experiments.NotifyCompletedExperiments(opts.Logger)
+
+				return nil
+			},
+		}),
+
+		// Strict Mode flags.
+
+		BoolWithDeprecatedFlag(opts, &cli.BoolFlag{
+			Name:    StrictModeFlagName,
+			EnvVars: EnvVars(StrictModeFlagName),
+			Usage:   "Enables strict mode for Terragrunt. For more information, see https://terragrunt.gruntwork.io/docs/reference/strict-mode .",
+			Setter: func(_ bool) error {
+				opts.StrictControls.EnableStrictMode()
+
+				return nil
+			},
+		}),
+		SliceWithDeprecatedFlag(opts, &cli.SliceFlag[string]{
+			Name:    StrictControlFlagName,
+			EnvVars: EnvVars(StrictControlFlagName),
+			Usage:   "Enables specific strict controls. For a list of available controls, see https://terragrunt.gruntwork.io/docs/reference/strict-mode .",
+			Setter:  opts.StrictControls.EnableControl,
+			Action: func(_ *cli.Context, _ []string) error {
+				opts.StrictControls.NotifyCompletedControls(opts.Logger)
+
+				return nil
+			},
+		}),
+
+		// Deprecated flags.
+
+		&cli.BoolFlag{
+			Name:        TerragruntDisableLogFormattingFlagName,
+			EnvVars:     EnvVars(TerragruntDisableLogFormattingFlagName),
+			Destination: &opts.DisableLogFormatting,
+			Usage:       "If specified, logs will be displayed in key/value format. By default, logs are formatted in a human readable format.",
+			Hidden:      true,
+			Action: func(_ *cli.Context, _ bool) error {
+				opts.LogFormatter.SetPlaceholders(format.NewKeyValueFormatPlaceholders())
+
+				newFlagName := LogCustomFormatFlagName + "=" + format.KeyValueFormatName
+
+				if err := opts.StrictControls.Evaluate(opts.Logger, strict.DeprecatedFlags, TerragruntDisableLogFormattingFlagName, newFlagName); err != nil {
```

Comment:
What's making this synchronize with the `EnableStrictMode` calls above? Do these flags get evaluated in series?